lib/stats/stats-query.c

```diff
@@ -243,7 +243,7 @@ _get_views(const gchar *filter)
   g_static_mutex_unlock(&stats_query_mutex);
 
   g_pattern_spec_free(pattern);
-  return views;
+  return g_list_reverse(views);
```

Comment:
The order is unspecified, as such reversing it is not really required here.

Reply:
Nice finding! I was thinking about this, and I would like to leave this as it is.

I agree the order is unspecified: it depends on the internal representation of the hashtables containing the counters. While the ordering in unspecified, it is deterministic. Removing this would reverse the output that users got used to. This is not a strong argument, and I am fine to discuss this separately. Personally, I do not really like that the output is not ordered. But for now, I  would like to maintain the original behavior.

Reply:
I would say that even a "keeping the ordering" is not true; as it is far being deterministic. Due to multi threading and the fact that the order of inserting a key has an effect on this listing. (sure with the above constructed syntactical config it is). As such user could not get used to an order... and should not either way.

Sure the gain with the lack of list reverse is way smaller, and the PR does improve speed as is, so I could approve.

(I would rather do `g_list_shuffle` instead of ordering it, just to make sure nobody relies on any ordering, see how much appending costs... as at the end of the day as a user I could do `syslog-ng-ctl stats|sort` that costs nothing to syslog-ng.)